instrumentation/jetty-httpclient/jetty-httpclient-12.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jetty/httpclient/v12_0/JettyHttpClient12Instrumentation.java

```diff
@@ -0,0 +1,124 @@
+import net.bytebuddy.matcher.ElementMatcher;
+import org.eclipse.jetty.client.Request;
+import org.eclipse.jetty.client.transport.HttpRequest;
+
+public class JettyHttpClient12Instrumentation implements TypeInstrumentation {
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("org.eclipse.jetty.client.transport.HttpRequest");
+  }
+
+  @Override
+  public void transform(TypeTransformer transformer) {
+    transformer.applyAdviceToMethod(
+        isMethod()
+            .and(named("send"))
+            .and(takesArgument(0, named("org.eclipse.jetty.client.Response$CompleteListener"))),
+        JettyHttpClient12Instrumentation.class.getName() + "$JettyHttpClient12SendAdvice");
+    // For request listeners
+    transformer.applyAdviceToMethod(
+        isMethod().and(nameContains("notify")),
+        JettyHttpClient12Instrumentation.class.getName() + "$JettyHttpClient12NotifyAdvice");
+  }
+
+  @SuppressWarnings("unused")
+  public static class JettyHttpClient12SendAdvice {
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void onEnterSend(
+        @Advice.This HttpRequest request,
+        @Advice.Local("otelContext") Context context,
+        @Advice.Local("otelScope") Scope scope) {
+      // start span
+      Context parentContext = Context.current();
+      JettyHttpClient12TracingInterceptor interceptor =
+          new JettyHttpClient12TracingInterceptor(parentContext, instrumenter());
+      interceptor.attachToRequest(request);
+      context = interceptor.getContext();
+      if (context == null) {
+        return;
+      }
+      // set context for responseListeners
+      VirtualField<Request, Context> virtualField = VirtualField.find(Request.class, Context.class);
+      virtualField.set(request, parentContext);
+      request.attribute(JETTY_CLIENT_CONTEXT_KEY, parentContext);
+
+      scope = context.makeCurrent();
+    }
+
+    @Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class)
+    public static void onExitSend(
+        @Advice.This HttpRequest request,
+        @Advice.Thrown Throwable throwable,
+        @Advice.Local("otelContext") Context context,
+        @Advice.Local("otelScope") Scope scope) {
+      if (scope == null) {
+        return;
+      }
+
+      // not ending span here unless error, span ended in the interceptor
+      scope.close();
+      if (throwable != null) {
+        instrumenter().end(context, request, null, throwable);
+      }
+    }
+  }
+
+  @SuppressWarnings("unused")
+  public static class JettyHttpClient12NotifyAdvice {
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void onEnterNotify(
+        @Advice.This HttpRequest request,
+        @Advice.Local("otelContext") Context context,
+        @Advice.Local("otelScope") Scope scope) {
+      if (!request.getAttributes().containsKey(JETTY_CLIENT_CONTEXT_KEY)) {
+        return;
+      }
```

Comment:
is this needed? Isn't returning on null context as done in the next block enough?

Reply:
Yes，it is done in the next block